demo/site/src/blocks/YouTubeVideoBlock.tsx

```diff
@@ -13,19 +13,23 @@ const getHeightInPercentForAspectRatio = (aspectRatio: YouTubeVideoBlockData["as
     }
 };
 
+const EXPECTED_YT_ID_LENGTH = 11;
+
+const parseYoutubeUrl = (url: string): { success: boolean; identifier?: string } => {
+    // regex from https://stackoverflow.com/a/27728417
+    const regExp = /^.*(?:(?:youtu\.be\/|v\/|vi\/|u\/\w\/|embed\/|shorts\/)|(?:(?:watch)?\?v(?:i)?=|&v(?:i)?=))([^#&?]*).*/;
+    const match = url.match(regExp);
+    const youtubeId = match && match[1].length == EXPECTED_YT_ID_LENGTH ? match[1] : null;
+
+    if (youtubeId) return { success: true, identifier: youtubeId };
+    else return { success: false };
+};
+
 const YouTubeVideoBlock: React.FunctionComponent<PropsWithData<YouTubeVideoBlockData>> = ({
     data: { youtubeIdentifier, autoplay, loop, showControls, aspectRatio },
 }) => {
-    try {
-        const url = new URL(youtubeIdentifier);
-        const searchParams = url.searchParams;
-        if (!searchParams.has("v")) {
-            throw new Error("URL has no ID (v) param");
-        }
-        youtubeIdentifier = searchParams.get("v") as string;
-    } catch (error) {
-        // no url, but ID was specified
-    }
+    const { success, identifier } = parseYoutubeUrl(youtubeIdentifier);
+    youtubeIdentifier = success && identifier ? identifier : "";
```

Comment:
`success` isn't really useful here, you can return identifier from `parseYouTubeUrl` directly